### PR TITLE
How To Auto-generate Admin Panels for Node.js with AdminBro

### PR DESCRIPTION
**Pull request by KayveTech**
### Article checklist
**NOTE: (Please ensure that you have only one open issue + linked pull request at a time. This will ensure that we complete the article in a timely manner from inception to publishing.)**
#### Preliminary Checks - Formatting and Structure 
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**
- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [x] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) -  [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker? We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**

### Contribution guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**

### CODE REVIEW

File: articles/nodejs-adminbro/index.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
+### How To Create an Admin Panel with Node.js and AdminBro
+
+AdminBro is an open-source package that adds an auto-generated admin interface to manage data in a single panel in Node.js applications during development. This adds the ability to find, monitor, and update the application data during development. Some of the features that AdminBro gives us include form validation, customization, configure role-based access control, and adding extra features to it. In this tutorial, we will set up a working admin interface that can be used to start managing application data in a Node.js application.
```
Comment (WanjaMIKE) [inline articles/nodejs-adminbro/index.md:3]: Add a link to AdminBro

File: articles/nodejs-adminbro/index.md [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+ - **express-session**: This session middleware will be required by our Admin panel.
+  
+On your terminal, run the command: 
+`npm install express mongoose express-formidable admin-bro`
+and the command
+`npm install @admin-bro/express @admin-bro/mongoose tslib express-session`
+ To use the AdminBro in various Node.js applications, check the [official documentation](https://adminbro.com/docs.html).
+
+
+#### Setting up the router middleware
```
Comment (WanjaMIKE) [inline articles/nodejs-adminbro/index.md:41]: Use only level three headings ```###```

File: articles/nodejs-adminbro/index.md [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    Databases: [],
+    rootPath: ‘/admin’,
+})
+
+const router = AdminBroExpress.buildRouter (adminBro)
+The next step is to set up the router as middleware using the Express.js app object:
+
+app.use(adminBro.options.rootPath, router)
+app.listen(8080, () => console.log(‘AdminBro is under localhost:8080/admin’))
+```
```
Comment (WanjaMIKE) [inline articles/nodejs-adminbro/index.md:60]: Could you explain the above code snippets.

File: articles/nodejs-adminbro/index.md [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+>>> Note: To implement this set up on an app that has an existing middleware stack, we need to make sure that AdminBro is on the
+>>>  first in the routing layers. This is because AdminBro cannot handle requests that have been transformed by other middleware.
+
+### Connecting the backend resources
+Now, we will seed the admin panel with data by establishing a connection to our database resources. This setup process should be repeated for each one of the ORMs that we draw application data from. In our case, we will use Mongoose to interact with the data
+### Summary
+Admin panel is a useful tool in. Happy coding!
```
Comment (WanjaMIKE) [inline articles/nodejs-adminbro/index.md:67]: Conclusion is too brief. Add more information .

Generally, add more infor to this article to make it more detailed and informative. For now, it's to brief for a technical article. Thanks

**Review by WanjaMIKE — CHANGES_REQUESTED** [review]
Please check the above suggestions. Thanks

**Comment by KayveTech** [comment]
Hey @WanjaMIKE it was initially a draft but I did implement the changes, kindly checkout and let me know of anything else needed.

**Comment by marienjus** [comment]
Hi, did you check my changes @WanjaMIKE

**Comment by WanjaMIKE** [comment]
@KayveTech why did you close this PR?

**Comment by KayveTech** [comment]
Done requested the changes that you requested :+1:

**Comment by WanjaMIKE** [comment]
Kindly, be patient. Am working on the PR?

**Comment by WanjaMIKE** [comment]
Nice article. I have a lot about creating Admin panels in Node.js using Admin bro

**Review by WanjaMIKE — APPROVED** [review]

**Comment by hectorkambow** [comment]
@KayveTech 👋 Good afternoon - and welcome to the EngEd program.
I'll be running a final review on your PR when ready.
Upon running your article through our 3rd party plagiarism checker it seemed to raise a few flags and the % was higher than we typically accept.
Please see attached PDF - and revisit the article to ensure we are contributing wholly unique and original content.
Be sure to see our resources page to see more info on [plagiarism ](https://www.scribbr.com/category/plagiarism/)and what is considered as such.

cc @WanjaMIKE

**Comment by KayveTech** [comment]
@hectorkambow no attached pdf?

**Comment by KayveTech** [comment]
@hectorkambow, I have the pliagiarism fixed. :+1:

**Comment by hectorkambow** [comment]
[2120420-dae06c4c0449fafaf46c-1620248741165.pdf](https://github.com/section-engineering-education/engineering-education/files/6430233/2120420-dae06c4c0449fafaf46c-1620248741165.pdf)

@KayveTech 

Sorry I forgot to attach the PDF

**Comment by KayveTech** [comment]
@hectorkambow I did fix the pliagiarism with quetext :+1:

**Comment by WanjaMIKE** [comment]
@KayveTech I noticed you changed your variables names. Did you update your repo?

`For the entire source code, check this [github repository](https://github.com/KayveTech/AdminBro-Express).`

**Comment by KayveTech** [comment]
Was kinda confusing having pliagiarism from variables and keywords, updated :+1:

**Comment by KayveTech** [comment]
@WanjaMIKE this was my first PR. It has the author details in it but the merged article I can't see. @hectorkambow is anything wrong about that?

**Comment by KayveTech** [comment]
The author details named start with a capital letter, didn't know about that, kindly edit the merged PR @hectorkambow .

**Comment by WanjaMIKE** [comment]
It's okay

**Pull request merged**